src/lib/table/table.md

```diff
@@ -1,50 +1,185 @@
-(e.g., sort headers) or next to the table (e.g. a paginator). Interactions that affect the
-rendered data (such as sorting and pagination) should be propagated through the table's data source.
+```ts
+this.myDataSource = new MatTableDataSource();
+this.myDataSource.data = dataToRender;
+```
 
+```html
+<mat-table [dataSource]=”myDataSource”>
+  ...
+</mat-table>
+```
+
+### Features
 
 #### Pagination
 
-The `<mat-paginator>` adds a pagination UI that can be used in conjunction with the `<mat-table>`. The
-paginator emits events that can be used to trigger an update via the table's data source.
+To paginate the table's data, add a `<mat-paginator>` after the `<mat-table>` and provide the 
+`MatPaginator` to the `MatTableDataSource`. The data source will automatically listen for page 
+changes made by the user and send the right paged data to the table.
+
+For more information on using and configuring the `<mat-paginator>`, check out the 
+[mat-paginator docs](https://material.angular.io/components/paginator/overview).
 
 <!-- example(table-pagination) -->
 
 #### Sorting
 
-Use the `matSort` directive and `<mat-sort-header>` adds a sorting UI the table's column headers. The
-sort headers emit events that can be used to trigger an update via the table's data source.
+To add sorting behavior to the table, add the `matSort` directive to the `<mat-table>` and add 
+`mat-sort-header` to each column header cell that should trigger sorting. Provide the `MatSort` 
+directive to the `MatTableDataSource` and it will automatically listen for sorting changes and 
+change the order of data rendered by the table.
+
+By default, the `MatTableDataSource` sorts with the assumption that the sorted column's name 
+matches the data property name that the column displays. For example, the following column 
+definition is named `position`, which matches the name of the property displayed in the row cell.
+
+```html
+<!-- Name Column -->
+<ng-container matColumnDef="position">
+  <mat-header-cell *matHeaderCellDef mat-sort-header> Name </mat-header-cell>
+  <mat-cell *matCellDef="let element"> {{element.position}} </mat-cell>
+</ng-container>
+```
+
+If the data properties do not match the column names, or if a more complex data property accessor 
+is required, then a custom `sortingDataAccessor` function can be set to override the default data 
+accessor on the `MatTableDataSource`.
 
 <!-- example(table-sorting) -->
 
+For more information on using and configuring the sorting behavior, check out the 
+[matSort docs](https://material.angular.io/components/sort/overview).
+
 #### Filtering
 
-While Angular Material does not offer a specific component for filtering tabular data, the table's
-data source can be updated based on any custom filter UI. Any filtering pattern need only trigger
-an update via the table's data source.
+To remove filtered rows from the table's data, simply provide the filter string to the 
```

Comment:
This section is too specifically about the `MatTableDataSource`. We should mention that we don't provide any filtering UI components and that filtering is up to the app. Then `MatTableDataSource` can be the "if you want something simple" extra thing.

Reply:
Flexibility is great, but this documentation is under angular/material2. Most developers won't have time to try a lot of different options. Would still be good to provide example code for MatTableDataSource.